### PR TITLE
build: postcss preset env support added

### PR DESCRIPTION
**Pull request by castastrophe**
## Description

The `postcss-preset-env` plugin offers support for modern CSS. It determines the polyfills needed based on targeted browsers. This project has a rich [feature list](https://github.com/csstools/postcss-plugins/blob/main/plugin-packs/postcss-preset-env/FEATURES.md) and active maintainer community.

- Remove legacy [dir] attribute selectors in favor of modern :dir usage in CSS
- Remove `&--<word>` syntax which is not supported in the current nesting spec
- Remove postcss-calc as a duplicate dependency
- Remove postcss-nested as a duplicate dependency

## How and where has this been tested?

Please tag yourself on the tests you've marked complete to confirm the tests have been run by someone other than the author.

### Validation steps

Expect no regressions in any components.

### Regression testing

Validate:

1. The documentation pages for at least two other components are still loading, including:

- [x] The pages render correctly, are accessible, and are responsive.

2. If components have been modified, VRTs have been run on this branch:

- [x] VRTs have been run and looked at.
- [x] Any VRT changes have been accepted (by reviewer and/or PR author), or there are no changes.

## To-do list

- [x] I have read the [contribution guidelines](/.github/CONTRIBUTING.md).
- [x] If my change impacts **other components**, I have tested to make sure they don't break.
- [x] ✨ This pull request is ready to merge. ✨

### CODE REVIEW

File: components/table/index.css [hunk 1]
```diff
@@ -110,7 +110,7 @@ governing permissions and limitations under the License.
   /* Passthrough for nested component(s) */
   --mod-thumbnail-size: var(--mod-table-thumbnail-size, var(--spectrum-table-thumbnail-size));
 
-	[dir="rtl"] &,
+	&:dir(rtl),
 	&:dir(rtl) {
```
Comment (mdt2) [inline components/table/index.css:114]: Duplicate here
```suggestion
	&:dir(rtl) {
```

File: components/treeview/index.css [hunk 2]
```diff
@@ -57,7 +57,7 @@ governing permissions and limitations under the License.
 
   --spectrum-treeview-indicator-animation-duration: var(--spectrum-animation-duration-100);
 
-	[dir="rtl"] &,
+	&:dir(rtl),
 	&:dir(rtl) {
```
Comment (mdt2) [inline components/treeview/index.css:61]:
```suggestion
	&:dir(rtl) {
```

**Review by mdt2 — APPROVED** [review]
This looks ok to me! As discussed in Slack, I picked a few components to test manually and they looked good, and your expectation is that there's enough RTL and hover state tests in VRTs to cover these changes 👍 Just found a couple duplicates.

File: components/table/index.css (hunk 1 above)
Comment (castastrophe) [reply components/table/index.css]: Great call-out - fixed this duplication in both instances and pushed up

**Pull request merged**